tests/contrib/mariadb/test_mariadb.py

```diff
@@ -0,0 +1,388 @@
+        END;"""
+    )
+
+    tracer.enabled = True
+    proc = "sp_sum"
+    data = (40, 2, None)
+    cursor.callproc(proc, data)
+
+
+    spans = tracer.pop()
+    assert spans, spans
+
+    # number of spans depends on mariadb implementation details,
+    # typically, internal calls to execute, but at least we
+    # can expect the last closed span to be our proc.
+    span = spans[len(spans) - 1]
+    assert_is_measured(span)
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.query"
+    assert span.span_type == "sql"
+    assert span.error == 0
+    assert span.get_metric("out.port") == 3306
+    assert_dict_issuperset(
+        span.meta,
+        {
+            "out.host": u"127.0.0.1",
+            "db.name": u"test",
+            "db.user": u"test",
+        },
+    )
+    assert span.get_tag("mariadb.query") is None
+
+def test_commit(connection, tracer):
+    connection.commit()
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.connection.commit"
+
+def test_rollback(connection, tracer):
+    connection.rollback()
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.connection.rollback"
+
+def test_analytics_default(connection, tracer):
+    cursor = connection.cursor()
+    cursor.execute("SELECT 1")
+    rows = cursor.fetchall()
+    assert len(rows) == 1
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) is None
+
+def test_analytics_with_rate(connection, tracer):
+    with override_config("mariadb", dict(analytics_enabled=True, analytics_sample_rate=0.5)):
+        cursor = connection.cursor()
+        cursor.execute("SELECT 1")
+        rows = cursor.fetchall()
+        assert len(rows) == 1
+        spans = tracer.pop()
+        assert len(spans) == 1
+        span = spans[0]
+        assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) == 0.5
+
+def test_analytics_without_rate(connection, tracer):
+    with override_config("mariadb", dict(analytics_enabled=True)):
+        cursor = connection.cursor()
+        cursor.execute("SELECT 1")
+        rows = cursor.fetchall()
+        assert len(rows) == 1
+        spans = tracer.pop()
+
+        assert len(spans) == 1
+
+        span = spans[0]
+        assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) == 1.0
+
+
+
+
+class TestMariadbPatch(TracerTestCase):
+    def setUp(self):
+        super(TestMariadbPatch, self).setUp()
+        patch()
+        self.connection = mariadb.connect(
+            user="test",
+            password="test",
+            host="127.0.0.1",
+            port=3306,
+            database="test"
+        )
+        Pin.override(self.connection, tracer=self.tracer)
+
+        @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
```

Comment:
we have another fixture you can use so you don't need to build a class/TracerTestCase

https://github.com/DataDog/dd-trace-py/blob/0dee2fcc7ecbdde7bb95b015b66b638660e349ef/tests/conftest.py#L22-L39